src/java.base/share/classes/com/sun/crypto/provider/SunJCE.java

```diff
@@ -99,24 +99,23 @@ public final class SunJCE extends Provider {
 
     private static boolean nativeChaCha20Init() {
         boolean nativeChaCha20 = true;
-        String nativeCryptTrace = GetPropertyAction.privilegedGetProperty("jdk.nativeCryptoTrace");
-        String nativeCryptStr = GetPropertyAction.privilegedGetProperty("jdk.nativeCrypto");
+        boolean nativeCryptTrace = NativeCrypto.isTraceEnabled();
 
-        if ((nativeCryptStr != null) && !Boolean.parseBoolean(nativeCryptStr)) {
-            /* nativeCrypto is explicitly disabled */
-            nativeChaCha20 = false;
-        } else {
-            String nativeChaCha20Str = GetPropertyAction.privilegedGetProperty("jdk.nativeChaCha20");
+        if (NativeCrypto.isEnabled()) {
+            String nativeChaCha20Str = GetPropertyAction.privilegedGetProperty("jdk.nativeChaCha20", "true");
```

Comment:
move that into NativeCrypto as well and expose it via a function as you did the others